recommender/static/js/src/cats.js

```diff
@@ -1,82 +1,202 @@
+    '.remove': '<span>Remove this resource and give the reason why you do that</span>',
+    '.previewingImg': '<span>Preview image (typically, a screenshot)</span>',
+    '.inTitle': '<span>Give a short (1-3 sentence) summary of the resource; ideally, this should be concise, but give enough detail to let students know whether this resources is useful to them</span>',
+    '.inUrl': '<span>Cut-and-paste the URL of the resource.</span>',
+    '.inDescriptionText': '<span>Give a paragraph of summary of the resource; the summary should be more detailed than you gave in Title</span>',
+    '.addResourceScreenshot': '<span>Upload a preview screenshot (in GIF/PNG/JPG) of the resource; ideally, this should let students know whether this resources is useful to them</span>',
+    '.editTitle': '<span>Give a short (1-3 sentence) summary of the resource; ideally, this should be concise, but give enough detail to let students know whether this resources is useful to them</span>',
+    '.editUrl': '<span>Cut-and-paste the URL of the resource.</span>', // TODO: Give instructions to go to element of learning sequence, or time in video
+    '.editDescriptionText': '<span>Give a paragraph of summary of the resource; the summary should be more detailed than you gave in Title</span>',
+    '.editResourceScreenshot': '<span>Upload a preview screenshot (in GIF/PNG/JPG) of the resource; ideally, this should let students know whether this resources is useful to them</span>',
+    '.backToViewButton': '<span>Go back to the main list</span>',
+    '.flagReason': '<span>Give a meaningful reason for why this resource should be removed</span>',
+    '.flagResource': '<span>Flag this resource as problematic and give your reason</span>',
+    '.deleteResource': '<span>Delete this resource</span>',
+    '.hideShow': '<span>Show a list of student-recommented related resources</span>',
+    '.hideShow.resourceListExpanded': '<span>Hide the recommendations list</span>',
+    '.checkIcon': '<span>Check the icon to endorse this resource</span>',
+    '.checkIcon.endorsed': '<span>This resource is endorsed by staff</span>',
+    '.resourceRankingForRemovalButton': '<span>Click to view resources for removal</span>',
+    '.resourceRankingForRemovalButton.removeMode': '<span>Click to view resources in ordinary decreasing-vote order</span>'
 };
 
-var uploadFileError = [
-    'FILE_TYPE_ERROR',
-    'IMPROPER_S3_SETUP',
-    '{"success": "Submission aborted!',
-    'FILE_SIZE_ERROR'
-];
-
 var uploadFileErrorText = {
-    'FILE_TYPE_ERROR': 'Please upload an image in GIF/JPG/PNG',
-    'IMPROPER_S3_SETUP': 'The configuration of Amazon S3 is not properly set',
-    '{"success": "Submission aborted!': 'Size of uploaded file exceeds threshold',
-    'FILE_SIZE_ERROR': 'Size of uploaded file exceeds threshold'
+    415: 'Please upload an image in GIF/JPG/PNG',
+    404: 'The configuration of Amazon S3 is not properly set',
+    413: 'Size of uploaded file exceeds threshold'
```

Comment:
This seems like the wrong place to match up error messages to HTTP responses. The server should include a message like this in the error response, so that the Javascript can display it to the user. These messages should be generated on the server because the same HTTP status code could be used for several different reasons. For example, a 404 could be due to the S3 configuration being unset, incorrectly set, or the configuration is fine and the requested file is simply not there. The client has no way to know -- only the server knows.